### PR TITLE
Switch occurrences of master to main

### PR DESCRIPTION
**Pull request by allisonchou**
Switches occurrences of master to main and removes master from the yml pipeline. Some links to other dotnet repos will be broken, but since all dotnet repos are scheduled to switch over to main this sprint, this should be temporary.
`eng/common` files have been left purposely untouched since it appears modifying them may break things. I also do not know when general Microsoft repos are scheduled to make the switch from master->main (e.g. Microsoft/azure-pipelines-tasks) so I left those links untouched for now as well.

### CODE REVIEW

File: tools/AnalyzerCodeGenerator/IncrementalGenerator/IncrementalGenerator.cs [hunk 1]
```diff
@@ -11,15 +11,15 @@ namespace AnalyzerCodeGenerator
 {
     public class IncrementalGenerator
     {
-        private static string _masterCsvFile;
+        private static string _mainCsvFile;
```
Comment (jmarolf) [inline tools/AnalyzerCodeGenerator/IncrementalGenerator/IncrementalGenerator.cs:14]: I think these should be reverted? They aren't referring the branch names

Comment (allisonchou) [reply tools/AnalyzerCodeGenerator/IncrementalGenerator/IncrementalGenerator.cs]: Done! Thanks.

**Review by jmarolf — APPROVED** [review]

File: docs/Analyzer Configuration.md [hunk 2]
```diff
@@ -208,7 +208,7 @@ Option Values: Names of null check validation methods (separated by `|`) that va
 Allowed method name formats:
 
   1. Method name only (includes all methods with the name, regardless of the containing type or namespace).
-  2. Fully qualified names in the [symbol's documentation ID format](https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#id-string-format)
+  2. Fully qualified names in the [symbol's documentation ID format](https://github.com/dotnet/csharplang/blob/main/spec/documentation-comments.md#id-string-format)
```
Comment (Youssef1313) [inline docs/Analyzer Configuration.md:211]: Is csharplang expected to move soon?

File: src/Microsoft.CodeAnalysis.Analyzers/Core/xlf/CodeAnalysisDiagnosticsResources.cs.xlf [hunk 3]
```diff
@@ -133,8 +133,8 @@
...
-        <target state="translated">Když se pro balíčky analyzátoru povolí sledování vydaných verzí, pomůže to sledovat a dokumentovat diagnostiky analyzátoru, které se dodávají a mění s každou jeho vydanou verzí. Podrobnosti najdete na adrese https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md.</target>
+        <source>Enabling release tracking for analyzer packages helps in tracking and documenting the analyzer diagnostics that ship and/or change with each analyzer release. See details at https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md.</source>
+        <target state="translated">Když se pro balíčky analyzátoru povolí sledování vydaných verzí, pomůže to sledovat a dokumentovat diagnostiky analyzátoru, které se dodávají a mění s každou jeho vydanou verzí. Podrobnosti najdete na adrese https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md.</target>
```
Comment (Youssef1313) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/xlf/CodeAnalysisDiagnosticsResources.cs.xlf:137]: I don't have enough information about the localization process. But I'm afraid this gets overwritten with the next loc pass. @wli3 Should the loc team be informed to update this in their database?

File: docs/Analyzer Configuration.md (hunk 2 above)
Comment (allisonchou) [reply docs/Analyzer Configuration.md:211]: Yep, I believe all dotnet repos are scheduled to make the switch this month, so this is just pre-emptive.

File: src/Microsoft.CodeAnalysis.Analyzers/Core/xlf/CodeAnalysisDiagnosticsResources.cs.xlf (hunk 3 above)
Comment (allisonchou) [reply src/Microsoft.CodeAnalysis.Analyzers/Core/xlf/CodeAnalysisDiagnosticsResources.cs.xlf:137]: Good point! Interested to hear from @wli3 whether we need to do anything here.

**Pull request merged**